File: app/src/main/java/me/simpleHook/platform/hook/entry/Lsp100HookEntry.java
package me.simpleHook.platform.hook.entry;

import static com.github.kyuubiran.ezxhelper.utils.HookUtilsKt.hookReturnConstant;
import static com.github.kyuubiran.ezxhelper.utils.MethodUtilsKt.findMethod;

import android.app.Application;
import android.content.pm.ApplicationInfo;
import androidx.annotation.NonNull;

import com.github.kyuubiran.ezxhelper.init.EzXHelperInit;

import java.lang.reflect.Method;

import io.github.libxposed.api.XposedModule;
import io.github.libxposed.api.annotations.XposedApiExact;
import io.github.qauxv.loader.sbl.common.ModuleLoader;
import io.github.qauxv.loader.sbl.lsp100.Lsp100HookImpl;
import io.github.qauxv.poststartup.StartupInfo;
import kotlin.jvm.functions.Function1;
import me.simpleHook.BuildConfig;
import me.simpleHook.platform.hook.utils.HookHelper;

/**
 * Entry point for libxpsoed API 100 (typically LSPosed).
 * <p>
 * The libxpsoed API is used as ART hook implementation.
 */
@XposedApiExact(100)
public class Lsp100HookEntry implements Lsp10xHookEntryHandler {

    private final XposedModule self;
    private XposedModule.ModuleLoadedParam mModule;

    /**
     * Instantiates a new Xposed module.
     * <p>
     * When the module is loaded into the target process, the constructor will be called.
     *
     * @param self  the Xposed module instance (this module)
     * @param param Information about the process in which the module is loaded
     */
    public Lsp100HookEntry(@NonNull XposedModule self, @NonNull XposedModule.ModuleLoadedParam param) {
        this.self = self;
        mModule = param;
        Lsp100HookImpl.init(self);
    }

    @XposedApiExact(100)
    public void onPackageLoaded(@NonNull XposedModule.PackageLoadedParam param) {
        EzXHelperInit.INSTANCE.initHandleLoadPackageLsp10x(param);
        StartupInfo.setHookBridge(Lsp100HookImpl.INSTANCE);
        StartupInfo.setLoaderService(Lsp100HookImpl.INSTANCE);

        HookEntry.INSTANCE.handleLoadPackage(param.getPackageName());
    }

}

File: app/src/main/java/me/simpleHook/hook/entry/Lsp100HookEntry.kt
package me.simpleHook.hook.entry

import android.app.Application
import android.content.Context
import android.util.Log
import com.github.kyuubiran.ezxhelper.init.EzXHelperInit
import com.github.kyuubiran.ezxhelper.utils.findMethod
import com.github.kyuubiran.ezxhelper.utils.hookAfter
import com.github.kyuubiran.ezxhelper.utils.hookReturnConstant
import io.github.libxposed.api.XposedInterface
import io.github.libxposed.api.XposedModule
import io.github.libxposed.api.XposedModuleInterface
import io.github.libxposed.api.XposedModuleInterface.ModuleLoadedParam
import io.github.qauxv.loader.sbl.lsp100.Lsp100HookImpl
import io.github.qauxv.poststartup.StartupInfo
import me.simpleHook.BuildConfig
import me.simpleHook.hook.util.HookHelper

class Lsp100HookEntry(base: XposedInterface, param: ModuleLoadedParam) : XposedModule(base, param) {

    init {
        Lsp100HookImpl.init(this)
    }

    override fun onPackageLoaded(param: XposedModuleInterface.PackageLoadedParam) {
        Log.d("SHOOK", param.packageName)
        EzXHelperInit.initHandleLoadPackageLsp100(param)
        StartupInfo.setHookBridge(Lsp100HookImpl.INSTANCE)
        StartupInfo.setLoaderService(Lsp100HookImpl.INSTANCE)

        if (param.packageName == BuildConfig.APPLICATION_ID) {
            findMethod("me.simpleHook.ui.activity.MainActivity") {
                name == "isModuleLive"
            }.hookReturnConstant(true)
        } else {
            if (HookHelper.isAppContextInitialized) return
            findMethod(Application::class.java) {
                name == "attach"
            }.hookAfter {
                HookHelper.initFields(
                    context = it.args[0] as Context,
                    param.packageName,
                    param.applicationInfo
                )
                HookHelper.appContext.getExternalFilesDirs(null)
                HookInit.startHook()
            }
        }
    }
}